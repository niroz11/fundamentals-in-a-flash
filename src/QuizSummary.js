import React, { Component } from 'react';

class QuizSummary extends Component {
  constructor(props) {
    super(props);
    this.state = {

    };
  }

  resetQuiz = (event) => {
    event.preventDefault();
    this.props.resetQuiz();
  }

  render() {
    let style = this.props.category.toLowerCase().split(' ').join('-')
    return (
      <div className={"summary-contain " + style}>
        <h2>{this.props.category} Quiz Summary</h2>
        <p>Great work!</p>
        <p>You answered {this.props.numberCorrect} out of {this.props.totalQuestions} questions correctly</p>
        <button onClick={this.resetQuiz}>Test Your Knowledge Again</button>
      </div>
    )
  }
}

export default QuizSummary;