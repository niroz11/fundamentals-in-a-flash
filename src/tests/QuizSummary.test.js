import React from 'react';
import QuizSummary from '../QuizSummary';
import { shallow } from 'enzyme';


describe('QuizSummary', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <QuizSummary
      />
    );
  });

  it('should match the snapshot with all data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  });
});