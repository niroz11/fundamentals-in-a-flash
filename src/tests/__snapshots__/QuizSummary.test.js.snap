// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuizSummary should match the snapshot with all data passed in 1`] = `undefined`;
